src/impl/icetransport.cpp

```diff
@@ -140,6 +140,10 @@ IceTransport::IceTransport(const Configuration &config, candidate_callback candi
 			addIceServer(server);
 }
 
+void IceTransport::setUfragAndPwd(string uFrag, string pwd) {
```

Comment:
Nitpick, but you could call this method `setIceAttributes` as it is the name of the underlying libjuice function.

You need to implement the method for libnice too, otherwise [the build is broken with libnice](https://github.com/paullouisageneau/libdatachannel/actions/runs/9462617852/job/26065990767?pr=1207#step:6:114). Ideally it should only print a warning like "Setting custom ICE ufrag and pwd is not supported with libnice".

Reply:
Done!

Reply:
Ah, so I've added the function, but it fails compilation with:

```
/home/runner/work/libdatachannel/libdatachannel/src/impl/icetransport.cpp: In member function ‘void rtc::impl::IceTransport::setIceAttributes(std::string, std::string)’:
/home/runner/work/libdatachannel/libdatachannel/src/impl/icetransport.cpp:576:44: error: unused parameter ‘uFrag’ [-Werror=unused-parameter]
  576 | void IceTransport::setIceAttributes(string uFrag, string pwd) {
      |                                     ~~~~~~~^~~~~
/home/runner/work/libdatachannel/libdatachannel/src/impl/icetransport.cpp:576:58: error: unused parameter ‘pwd’ [-Werror=unused-parameter]
  576 | void IceTransport::setIceAttributes(string uFrag, string pwd) {
      |                                                   ~~~~~~~^~~
[ 52%] Building CXX object CMakeFiles/datachannel.dir/src/impl/init.cpp.o
```

If I remove the unused parameters to make the compiler happy, it starts to fail with:

```
/home/runner/work/libdatachannel/libdatachannel/src/impl/icetransport.cpp:576:6: error: no declaration matches ‘void rtc::impl::IceTransport::setIceAttributes()’
  576 | void IceTransport::setIceAttributes() {
      |      ^~~~~~~~~~~~
In file included from /home/runner/work/libdatachannel/libdatachannel/src/impl/icetransport.cpp:9:
/home/runner/work/libdatachannel/libdatachannel/src/impl/icetransport.hpp:54:14: note: candidate is: ‘void rtc::impl::IceTransport::setIceAttributes(std::string, std::string)’
   54 |         void setIceAttributes(string uFrag, string pwd);
      |              ^~~~~~~~~~~~~~~~
/home/runner/work/libdatachannel/libdatachannel/src/impl/icetransport.hpp:33:7: note: ‘class rtc::impl::IceTransport’ defined here
   33 | class IceTransport : public Transport {
      |       ^~~~~~~~~~~~
```

My C++ is weak - do you know how I can get this to work?

Reply:
All good, I removed the parameter names.